cmd/cosign/cli/attest/common.go

```diff
@@ -33,3 +37,15 @@ func predicateReader(predicatePath string) (io.ReadCloser, error) {
 	}
 	return f, nil
 }
+
+func getEnvelopeSigBytes(envelopeBytes []byte) ([]byte, error) {
+	var envelope dsse.Envelope
+	err := json.Unmarshal(envelopeBytes, &envelope)
+	if err != nil {
+		return nil, err
+	}
+	if len(envelope.Signatures) == 0 {
+		return nil, fmt.Errorf("envelope has no signatures")
+	}
+	return base64.StdEncoding.DecodeString(envelope.Signatures[0].Sig)
```

Comment:
Since cosign produced this signature, we know that it should have exactly one. There are several other places we assume there is only one signature in a DSSE, [including in sigstore-go](https://github.com/sigstore/sigstore-go/blob/main/pkg/bundle/signature_content.go#L100). I'm okay without checking in this context, but we might want to add some validation in the sigstore-go verifier (in the linked file) that will warn/err if there are multiple present when not expected.